core/src/test/resources/sql/struct.iq

```diff
@@ -47,4 +47,40 @@ select distinct * from (values
 
 !ok
 
+# [CALCITE-3482] Equality of nested ROWs returns false for identical literal value
+select * from
+(values
+    (1, ROW(1,1)),
+    (2, ROW(2,2)),
+    (3, ROW(1,1)),
+    (4, ROW(2,2))) as t(id,struct)
+where t.struct = ROW(2,2);
++----+--------+
+| ID | STRUCT |
++----+--------+
+|  2 | {2, 2} |
+|  4 | {2, 2} |
++----+--------+
+(2 rows)
+
+!ok
+
+# [CALCITE-3482] Equality of nested ROWs returns false for identical literal value
+select * from
+(values
+    (1, ROW(2, ROW(4,2))),
+    (2, ROW(2, ROW(4,4))),
+    (3, ROW(2, ROW(1,4))),
+    (4, ROW(2, ROW(4,4)))) as t(id,struct)
+where t.struct = ROW(2, ROW(4,4));
```

Comment:
Can we give them different values, e.g. (3,4) and (4,3)? In order to verify that the ordering does matter.

Reply:
Sure, done.